samples/kubernetes/deploy-aks.sh

```diff
@@ -0,0 +1,214 @@
+then
+    # There is a problem with the automatic SP creation in az aks create, so we will make the SP first
+    # https://github.com/Azure/azure-cli/issues/9585
+
+    # First let's see if we have it service principal already
+    spId=""
+    spPass=""
+
+    clientIdKeyvaultSecret=`az keyvault secret show --vault-name $keyvaultName --name service-principal-client-id 2> /dev/null || true`
+    clientSecretKeyvaultSecret=`az keyvault secret show --vault-name $keyvaultName --name service-principal-client-secret 2> /dev/null || true`
+
+    if [ -n "$clientIdKeyvaultSecret" ] && [ -n "$clientSecretKeyvaultSecret" ]; then
+        spId=`echo $clientIdKeyvaultSecret | jq -r .value`
+        spPass=`echo $clientSecretKeyvaultSecret | jq -r .value`
+    else
+        subscription=$(az account show | jq -r .id)
+        sp=$(az ad sp create-for-rbac --scope /subscriptions/${subscription}/resourceGroups/${resourceGroupName} --role Contributor --output json)
+        spId=$(echo $sp | jq -r '.appId')
+        spPass=$(echo $sp | jq -r '.password')
+
+        az keyvault secret set --name "service-principal-client-id" --vault-name $keyvaultName --value $spId
+        az keyvault secret set --name "service-principal-client-secret" --vault-name $keyvaultName --value $spPass
+
+        sleep 30 # This is necessary make sure the service principal is available for the AKS deployment.
+    fi
+
+    az aks create \
+        --resource-group $resourceGroupName \
+        --name $clusterName \
+        --node-count 1 \
+        --node-vm-size $vmSize \
+        --vm-set-type VirtualMachineScaleSets \
+        --load-balancer-sku standard \
+        --enable-cluster-autoscaler \
+        --min-count 1 \
+        --max-count $maxNodes \
+        --service-principal $spId \
+        --client-secret $spPass \
+        --ssh-key-value ${sshKeyFile}.pub \
+        --kubernetes-version 1.18.2
+else
+    az aks update \
+        --resource-group $resourceGroupName \
+        --name $clusterName \
+        --update-cluster-autoscaler \
+        --min-count 1 \
+        --max-count $maxNodes
+fi
+
+# Get credentials for kubectl
+az aks get-credentials --name $clusterName --resource-group $resourceGroupName
+
+# Add the official stable repository
+helm repo add stable https://kubernetes-charts.storage.googleapis.com/
+
+# Install nginx
+foundIngressControllerNamespace=`kubectl get namespace -o json | jq '.items[] | select(.metadata.name == "ingress-controller")'`
+if [ -z "$foundIngressControllerNamespace" ]; then
+    kubectl create namespace ingress-controller
+fi
+
+# The specification of linux nodes here is not really necessary since that is all we have
+# but if we add Windows nodes in the future, we will want to make sure nginx is on Linux
+helm upgrade --install nginx-ingress stable/nginx-ingress \
+    --namespace ingress-controller \
+    --set controller.replicaCount=2 \
+    --set controller.nodeSelector."beta\.kubernetes\.io/os"=linux \
```

Comment:
Looks like this label has been deprecated. https://kubernetes.io/docs/reference/kubernetes-api/labels-annotations-taints/#betakubernetesioos-deprecated

Reply:
also is the quoting necessary?

Reply:
Removed beta. quotes and backslashes are needed since the name of the field is `kubernetes.io/os` if quotes and backslashes are not provided it is interpreted has:
```
controller:
   nodeSelector:
      beta:
         kubernetes/os:
```